src/compat/eme/custom_media_keys/index.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Observable,
+  of as observableOf,
+  throwError as observableThrow,
+} from "rxjs";
+import castToObservable from "../../../utils/cast_to_observable";
+import {
+  // XXX TODO remove when the issue is resolved
+  // https://github.com/Microsoft/TypeScript/issues/19189
+  ICompatMediaKeySystemAccess,
+  ICompatMediaKeySystemConfiguration,
+  MediaKeys_,
+} from "../../browser_compatibility_types";
+import { isIE11 } from "../../browser_detection";
+import isNode from "../../is_node";
+import shouldUseWebKitMediaKeys from "../../should_use_webkit_media_keys";
+import CustomMediaKeySystemAccess from "./../custom_key_system_access";
+import getIE11MediaKeysCallbacks from "./ie11_media_keys";
+import getOldKitWebKitMediaKeyCallbacks, {
+  isOldWebkitMediaElement
+} from "./oldwebkit_media_keys";
+import {
+  ICustomMediaKeys,
+  ICustomMediaKeySession,
+} from "./types";
+import getWebKitMediaKeysCallbacks from "./webkit_media_keys";
+
+/* tslint:disable no-unsafe-any */
+const { MSMediaKeys,
+        WebKitMediaKeys } = (window as any);
+/* tslint:enable no-unsafe-any */
+
+let isTypeSupported = (keyType: string): boolean => {
+  if ((MediaKeys_ as any).isTypeSupported === undefined) {
+    throw new Error("No isTypeSupported on MediaKeys.");
+  }
+  /* tslint:disable no-unsafe-any */
+  return (MediaKeys_ as any).isTypeSupported(keyType);
+  /* tslint:enable no-unsafe-any */
+};
+let createCustomMediaKeys = (keyType: string) => {
```

Comment:
All variables declared until now seem to only be used in the `else` condition below. Why not define then in that condition?

Reply:
Fixed